projects/packages/external-connections/src/class-external-connections.php

```diff
@@ -0,0 +1,272 @@
+	 */
+	public static function get_connection( $service ) {
+		if ( ( new Host() )->is_wpcom_simple() ) {
+			require_lib( 'external-media-service/external-media-list' );
+			require_lib( 'external-connections' );
+
+			$connections = \WPCOM_External_Connections::init();
+			$token       = \ExternalMediaService::get_service_token( $service, get_current_user_id() );
+
+			if ( ! empty( $token->unique_id ) ) {
+				return $connections->get_keyring_connection_item( $token->unique_id );
+			}
+		} else {
+			$response = Client::wpcom_json_api_request_as_user( '/me/connections' );
+			if ( is_wp_error( $response ) ) {
+				return null;
+			}
+			$body = json_decode( wp_remote_retrieve_body( $response ) );
+			if ( isset( $body->connections ) && is_array( $body->connections ) ) {
+				foreach ( $body->connections as $connection ) {
+					if ( $service === $connection->service ) {
+						return (array) $connection;
+					}
+				}
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Checks if a connection to the specified service exists and is active.
+	 *
+	 * @param string $service The service identifier.
+	 *
+	 * @return bool True if an active connection exists, false otherwise.
+	 */
+	public static function has_connection( $service ) {
+		$connection = self::get_connection( $service );
+
+		if ( empty( $connection ) ) {
+			return false;
+		}
+
+		$connection_status = $connection['status'] ?? 'not_connected';
+		return $connection_status === 'ok';
+	}
+
+	/**
+	 * Deletes a connection for the provided service.
+	 *
+	 * @param string $service The service identifier.
+	 */
+	public static function delete_connection( $service ) {
+		$connection = self::get_connection( $service );
+		if ( empty( $connection ) ) {
+			return;
+		}
+
+		if ( ( new Host() )->is_wpcom_simple() ) {
+			if ( get_current_user_id() === $connection['user_ID'] ) {
+				require_lib( 'external-connections' );
+				$connections = \WPCOM_External_Connections::init();
+				$connections->delete_keyring_connection( $connection['ID'] );
+			}
+		} else {
+			Client::wpcom_json_api_request_as_user(
+				'/me/connections/' . $connection['ID'],
+				'2',
+				array( 'method' => 'DELETE' )
+			);
+		}
+	}
+
+	/**
+	 * Registers connection settings for the provided services on a specified settings page.
+	 *
+	 * @param string $page The slug of the settings page where the connection settings should be added.
+	 * @param array  $services A list of services to be configured, where each service contains 'service', 'title',
+	 *                          'description', and 'support_link' keys.
+	 */
+	public static function add_connections_settings_section_and_fields( $page, $services ) {
+		global $pagenow;
+
+		if ( $pagenow !== "options-$page.php" ) {
+			return;
+		}
+
+		$host = new Host();
+
+		if ( ! $host->is_wpcom_simple() ) {
+			$connection = new Connection_Manager( 'jetpack' );
+			$status     = new Status();
+
+			if ( $status->is_offline_mode() ) {
```

Comment:
Nit: Can we combine all three earily returns? something like $status->is_offline_mode() || ! $connection->has_connected_owner() || ! $connection->is_user_connected()

Since it's using an or, it would shortcircuit the expression from left to right; it won't execute the method if one of the left is true.

Reply:
252243be91ac983cb1c636b0a71d4f677a5ee495